src/api/dependency-discovery/src/dependency-list.js

```diff
@@ -0,0 +1,11 @@
+const { readFile } = require('fs/promises');
+const { join } = require('path');
+const { cwd } = require('process');
+
+async function getDependencies() {
+  const dependencies = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+
+  return [...new Set(dependencies.split('\n').filter((line) => line !== ''))];
+}
+
+module.exports = getDependencies();
```

Comment:
Because the `await` keyword is not valid in the top level of a file. When you run the project with `pnpm start`, it will show this error:

```
SyntaxError: await is only valid in async functions and the top level bodies of modules
```

That is why we have to annoyingly await every time we access the object.